tiledb/sm/query/readers/dense_reader.cc

```diff
@@ -368,7 +368,7 @@ Status DenseReader::dense_read() {
   // This is as far as we should go before implementing this properly in a task
   // graph, where the start and end of every piece of work can clearly be
   // identified.
-  ThreadPool::Task compute_task;
+  ThreadPool::SharedTask compute_task;
```

Comment:
It's easy to see that this is waited upon multiple times - but if it worked fine before, why is it necessary for it to be shared now?